.github/workflows/windows-build-xpu.yml

```diff
@@ -0,0 +1,128 @@
+    VS2022INSTALLDIR: 'C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools'
+
+jobs:
+  Preparation:
+    runs-on: ["${{ inputs.win-runner }}"]
+    steps:
+      - name: Checkout PyTorch
+        uses: pytorch/pytorch/.github/actions/checkout-pytorch@main
+        with:
+          no-sudo: true
+
+      - name: Install oneAPI bundle
+        shell: cmd
+        run: |
+          if exist "C:\Program Files (x86)\Intel\oneAPI\setvars.bat" (
+            echo "oneAPI bundle already installed"
+          ) else (
+            echo "setvars.bat not found, installing oneAPI bundle"
+            call .ci\pytorch\win-test-helpers\installation-helpers\install_oneapi_bundle.bat
+            echo "oneAPI bundle installed"
+          )
+    
+      - name: Set conda environment
+        shell: cmd
+        run: |
+          call %conda_path%\Scripts\activate.bat
+          call conda remove -n %conda_env% --all -y
+          call conda create -n %conda_env% python=%python_version% -y
+          call conda activate %conda_env%
+          call conda install cmake==3.26.4 ninja -y
+          call pip install pyyaml
+          call conda list
+
+  Build-whls:
+    runs-on: ["${{ inputs.win-runner }}"]
+    needs: [Preparation]
+    steps:
+     - name: Build Torch
+       shell: cmd
+       run: |
+         call %conda_path%\Scripts\activate.bat %conda_env%
+         call "C:\Program Files (x86)\Intel\oneAPI\setvars.bat"
+         if exist "pytorch" (
+           rmdir /s /q "pytorch"
+         )
+         git clone https://github.com/pytorch/pytorch.git pytorch > git_pytorch_repo_log.txt
+         cd pytorch
+         call conda install conda-forge::rust -y
+         call conda install typing_extensions -y
+         python -m pip install -r requirements.txt > pip_install_pytorch_requirements_log.txt
+         python setup.py bdist_wheel > build_torch_wheel_log.txt
+
+  Upload-artifacts:
+    runs-on: ["${{ inputs.win-runner }}"]
+    needs: [Preparation,Build-whls]
+    steps:
+    - name: Upload pytorch wheel
+      uses: actions/upload-artifact@v2
+      with:
+        name: pytorch wheel
+        path: 'pytorch\dist'
+
+    - name: Upload torchvision wheel
+      uses: actions/upload-artifact@v2
+      with:
+        name: torchvision wheel
+        path: 'vision\dist'
+
+    - name: Upload logs
+      uses: actions/upload-artifact@v2
+      with:
+        name: logs
+        path: |
+          pytorch\git_pytorch_repo_log.txt
+          pytorch\pip_install_pytorch_requirements_log.txt
+          pytorch\build_torch_wheel_log.txt
+          vision\git_vision_repo_log.txt
+          vision\pip_install_vision_requirements_log.txt
+          vision\build_vision_wheel_log.txt
+
+    - name: Store PyTorch Build Artifacts on S3
+      uses: seemethere/upload-artifact-s3@v5
+      with:
+        name: ${{ inputs.build-environment }}
+        retention-days: 14
+        if-no-files-found: error
+        path: |
+          pytorch\dist\torch-*.whl
```

Comment:
Do existing runners upload binaries?

Reply:
Yes, they put all things in a zip file and upload the zip file. I am working on to see if I can directly use their existing build configurations.